src/app/asset-upload/asset-upload.component.ts

```diff
@@ -1,19 +1,54 @@
 import {Component} from '@angular/core';
 import {AssetUploadService} from './asset-upload.service';
+import {EventEmitter} from 'protractor';
 
 @Component({
   selector: 'app-asset-upload',
   templateUrl: './asset-upload.component.html',
   styleUrls: ['./asset-upload.component.scss'],
 })
 export class AssetUploadComponent {
-  file: File | null = null;
+  assetLink = '';
+  isSizeValid = true;
+  fileName = null;
 
   constructor(private service: AssetUploadService) {}
 
+  assetLinkUpload(assetLink: string) {
+    // Fetch the asset and parse the response stream as a blob
+    fetch(assetLink)
+      .then(response => response.blob())
+      .then(blob => {
+        const filename = assetLink.substring(assetLink.lastIndexOf('/'));
+        const file = new File([blob], filename, {type: blob.type});
+        this.updateAssets(assetLink, file);
+        this.service.updateIsAssetLink(true);
+        this.fileName = '';
+      });
+  }
+
   onFileInput(fileInput: any) {
-    this.file = fileInput.target.files[0];
-    const assetSrc = URL.createObjectURL(this.file);
-    this.service.updateAssets(assetSrc, this.file);
+    const file = fileInput.target.files[0];
+    const assetSrc = URL.createObjectURL(file);
+    this.updateAssets(assetSrc, file);
+    this.service.updateIsAssetLink(false);
+    this.fileName = file.name;
+    this.assetLink = '';
+
+    // addresses the case where user uploads the same file in succession
+    fileInput.target.value = '';
+  }
+
+  private updateAssets(src: string, file: File) {
+    this.validateSize(file);
+    if (this.isSizeValid) {
+      this.service.updateAssets(src, file);
+    } else {
+      this.service.updateAssets('', null);
+    }
+  }
+
+  private validateSize(file: File) {
+    this.isSizeValid = !(file.type.includes('video') && file.size > 4000000);
```

Comment:
Assign the 4000000 to a constant and reference the constant here. Otherwise it's not obvious what this number represents (like what unit does it refer to?)

```suggestion
    this.isSizeValid = !(file.type.includes('video') && file.size > FOUR_MB);
```